lib/facil/http/http1.c

```diff
@@ -342,6 +345,22 @@ static int http1_http2websocket_server(http_s *h, websocket_settings_s *args) {
     goto bad_request;
   stmp = fiobj_obj2cstr(tmp);
 
+  FIOBJ tmp_protocol = fiobj_hash_get2(h->headers, sec_protocol);
+  if (tmp_protocol) {
+    char *protocol_str = fiobj_obj2cstr(tmp_protocol).data;
+
+    if (strlen(protocol_str)) {
+      printf("HALLO\n");
+      char *tmp_ptr;
+          
+      tmp_ptr = strchr(protocol_str, ',');
+      if (tmp_ptr != NULL)
+          *tmp_ptr = '\0';
```

Comment:
Although I agree that it's better to avoid an additional string copy, I think this is destructive to the original copy which someone might still want to access later on.

It is better to use `tmp_protocol = fiobj_str_new(protocol_str, (tmp_ptr - protocol_str))` instead of destructively add a NUL char just to recalculate the length (especially since the length is already known by the value of `tmp_ptr`.